server/main.js

```diff
@@ -123,40 +125,43 @@ Meteor.publish('tabular_getInfo', function (tableName, selector, sort, skip, lim
     const paths = getSearchPaths(table);
     const newSort = transformSortArray(findOptions.sort);
 
-    filteredRecordIds = table.searchCustom(
+    filteredRecordIdsPromise = Promise.resolve( table.searchCustom(
       this.userId,
       newSelector,
       tokens,
       paths,
       newSort,
       skip,
       limit
-    );
+    ));
   } else {
     filteredCursor = table.collection.find(newSelector, findOptions);
-
-    filteredRecordIds = filteredCursor.map((doc) => doc._id);
+    filteredRecordIdsPromise = filteredCursor.mapAsync((doc) => doc._id);
```

Comment:
Is it worth making the top-level publish async so this could be an await and simplify all the resultant code ?

Reply:
Yes, indeed ! I didn't even thought to that :(